### PR TITLE
feat(@types/node): add `stream/consumers`

### PR DESCRIPTION
**Pull request by favna**
This is a set of new functions added in Node 16.7. As the types are now 16.7.x, we can add them.

NodeJS docs: https://nodejs.org/dist/latest-v16.x/docs/api/webstreams.html#webstreams_utility_consumers

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v16.x/docs/api/webstreams.html#webstreams_utility_consumers
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

### CODE REVIEW

**Comment by Favna** [comment]
Failed CI doesn't seem to be related to my changes so I'm going to need a second opinion on this.

**Comment by Favna** [comment]
***Note***

For some bizarre reason the `index.d.ts` has this comment:

> `// NOTE: These definitions support NodeJS and TypeScript 3.7.`.

But the comment of `// Minimum TypeScript Version: 3.7` was completely missing. 

Even more bizarre, there were folders of `ts3.6` which caused dtslint to try and use v3.6 which in turn caused the test framework setup to completely fail as seen in this workflow run: https://github.com/DefinitelyTyped/DefinitelyTyped/runs/3395665069?check_suite_focus=true
```
=== ERRORS ===



Error in node
AssertionError [ERR_ASSERTION]: Expected "actual" to be strictly unequal to: -1
    at next (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:195:12)
    at Array.map (<anonymous>)
    at /home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:167:92
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
```

I have moved all the `3.6` to `3.7` and added the `Minimum TypeScript Version` comment and now CI is properly green.

***This problem also exists on the master branch, I ensured to be up-to-date with the DefinitelyTyped master branch and tested there and got the same error***

**Comment by SimonSchick** [comment]
ts3.6 is being phased out, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54895

You can safely ignore these errors and we can proceed once this PR is submitted.

**Comment by Favna** [comment]
> ts3.6 is being phased out, see #54895
> 
> You can safely ignore these errors and we can proceed once this PR is submitted.

So should I revert those changes for now then? Or should I just keep them for now and rebase after https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54895 is merged (which I'll have to do regardless)

**Comment by SimonSchick** [comment]
I'd revert them as you won't really have to do any of it.

**Comment by Favna** [comment]
> I'd revert them as you won't really have to do any of it.

Done. Hopefully your PR will get merged in the near future.

**Comment by SimonSchick** [comment]
I anticipate it be merged before end of week.

**Review by LinusU — APPROVED** [review]
Nice! 👍

**Review by peterblazejewicz — APPROVED** [review]

File: types/node/stream/consumers.d.ts
Comment (alecgibson) [inline types/node/stream/consumers.d.ts:13]: I'm not sure this comment is accurate? It looks like this interface is leaking into global definitions:

149748342-48857820-a09b-439a-a7c7-f954d02f1aab

I can't remember where I read it (maybe in a linter rule or something?), but I'm pretty sure _all_ interfaces in a `.d.ts` file are exported, so that consumers have access to references to them. Very happy to be proven wrong on this point, though.

Comment (thw0rted) [reply types/node/stream/consumers.d.ts:13]: Good news, the correct types have been added in the interim so this erroneous "duplicate" is no longer needed.  I just opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59905 which I think will resolve this error.

Comment (favna) [reply types/node/stream/consumers.d.ts:13]: @thw0rted With the benefit of the doubt I will think you didn't mean it as such but I feel called out here anyway, and so I feel a need to defend myself.

I specifically did *not* add myself to the contributors list for node types because I have no interest in being pinged for reviews and I have hardly any interest in Node's types to begin with as long as they work, and this has for many months. 

Furthermore, even though yes it might be wrong that I made this change, that change was made in a pull request just like your own. That means it was reviewed and it was merged by a DT maintainer. Simply denotsting that I did bad here is therefore coming over as shallow because clearly other people were involved as well.

Comment (thw0rted) [reply types/node/stream/consumers.d.ts:13]: You're right that I didn't mean it as a personal attack -- reading back through my comment I can definitely see how you'd take it that way, and I'm sorry.  (I can sympathize about not signing yourself up as a contributor, the scope is unlike anything else on DT!)

The situation is very confusing, and I don't know that anybody can really be blamed directly -- Node chose to add this partial implementation of some web APIs, and worse they're reusing web class names for incomplete implementations that aren't available at the same scope.  When you made your PR you didn't even have a "Node Blob" to reference, so your approach was totally reasonable; bleeding into global was an honest mistake.

To make matters worse, I don't see any way to write a test (or I guess a "meta" test?) that would prevent this.  Maybe the package.json or tsconfig file could have a field with a list of globals introduced by the package, and if anything else bleeds out, the test would fail?  I don't really know enough about TS internals to suggest how to implement something like that but it might be worth kicking the idea around in Discussions -- I'll take a look.

**Pull request merged**
